workflow/common/util.go

```diff
@@ -263,6 +263,18 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.ArgumentsProvider, globalParams,
 		if argParam != nil && argParam.Value != nil {
 			inParam.Value = argParam.Value
 		}
+
+		if inParam.ValueFrom != nil && inParam.ValueFrom.ConfigMapKeyRef != nil {
```

Comment:
we need to be careful here, this could be an expensive operation that only needs to fail once to fail the workflow - it might be about time we had a config map informer - @sarabala1979